src/functionalTest/java/uk/gov/hmcts/dts/fact/admin/AdminCourtPostcodeEndpointTest.java

```diff
@@ -0,0 +1,257 @@
+    }
+
+    /************************************************************* POST request tests section. ***************************************************************/
+    @Test
+    public void shouldCreateValidPostcodes() throws JsonProcessingException {
+
+        final List<String> currentPostcodes = getCurrentPostcodes();
+        final String updatedJson = objectMapper().writeValueAsString(POSTCODES_VALID);
+        final var response = doPostRequest(
+            BIRMINGHAM_COURT_POSTCODES_PATH,
+            Map.of(AUTHORIZATION, BEARER + superAdminToken),
+            updatedJson
+        );
+        assertThat(response.statusCode()).isEqualTo(CREATED.value());
+        final List<String> updatedCourtPostcodes = response.body().jsonPath().getList(
+            ".",
+            String.class
+        );
+        final List<String> trimmedPostcodes = POSTCODES_VALID.stream().map(x -> x.replaceAll(" ", "")).collect(Collectors.toList());
+
+        assertThat(updatedCourtPostcodes).containsExactlyElementsOf(trimmedPostcodes);
+
+        //clean up by removing added record
+        final var cleanUpResponse = doDeleteRequest(
+            BIRMINGHAM_COURT_POSTCODES_PATH,
+            Map.of(AUTHORIZATION, BEARER + superAdminToken),
+            updatedJson
+        );
+        assertThat(cleanUpResponse.statusCode()).isEqualTo(OK.value());
+        assertThat(cleanUpResponse.getBody().as(int.class)).isEqualTo(POSTCODES_VALID.size());
+
+        final var responseAfterClean = doGetRequest(
+            BIRMINGHAM_COURT_POSTCODES_PATH,
+            Map.of(AUTHORIZATION, BEARER + authenticatedToken)
+        );
+
+        final List<String> cleanCourtPostcodes = responseAfterClean.body().jsonPath().getList(
+            ".",
+            String.class
+        );
+        assertThat(cleanCourtPostcodes).containsExactlyElementsOf(currentPostcodes);
+    }
+
+    @Test
+    public void shouldNotCreateDuplicatePostcodes() throws JsonProcessingException {
+
+        final List<String> currentPostcodes = getCurrentPostcodes();
+        final String updatedJson = objectMapper().writeValueAsString(POSTCODES_DUPLICATE);
+        final var response = doPostRequest(
+            BIRMINGHAM_COURT_POSTCODES_PATH,
+            Map.of(AUTHORIZATION, BEARER + superAdminToken),
+            updatedJson
+        );
+        assertThat(response.statusCode()).isEqualTo(CREATED.value());
+        final List<String> updatedCourtPostcodes = response.body().jsonPath().getList(
+            ".",
+            String.class
+        );
+
+        //Checking only unique values updated in database
+        final List<String> trimmedPostcodes = POSTCODES_DUPLICATE.stream().map(x -> x.replaceAll(" ", "")).collect(Collectors.toList());
+        final Set<String> set = new TreeSet<>(trimmedPostcodes);
+        assertThat(updatedCourtPostcodes.size()).isEqualTo(set.size());
+        assertThat(trimmedPostcodes).containsAll(updatedCourtPostcodes);
+
```

Comment:
@ashwini-mv  I think the 4 lines above could be simplified to below. If you use containsExactlyElementsOf(), it should be checking for size as well so don't need an explicit for size here.

final Set<String> trimmedPostcodes = POSTCODES_DUPLICATE.stream().map(x -> x.replaceAll(" ", "")).collect(Collectors.toSet());
assertThat(updatedCourtPostcodes).containsExactltyElementsOf(trimmedPostcodes)

Reply:
Done